near-contract-standards/src/fungible_token/implementation.rs

```diff
@@ -0,0 +1,256 @@
+use crate::fungible_token::core::FungibleTokenCore;
+use crate::fungible_token::resolver::FungibleTokenResolver;
+use crate::storage_manager::{AccountStorageBalance, StorageManager};
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::LookupMap;
+use near_sdk::json_types::{ValidAccountId, U128};
+use near_sdk::{
+    assert_one_yocto, env, ext_contract, log, AccountId, Balance, Gas, Promise, PromiseResult,
+    StorageUsage,
+};
+
+const GAS_FOR_RESOLVE_TRANSFER: Gas = 5_000_000_000_000;
+const GAS_FOR_FT_TRANSFER_CALL: Gas = 25_000_000_000_000 + GAS_FOR_RESOLVE_TRANSFER;
+
+const NO_DEPOSIT: Balance = 0;
+
+#[ext_contract(ext_self)]
+trait FungibleTokenResolver {
+    fn ft_resolve_transfer(
+        &mut self,
+        sender_id: AccountId,
+        receiver_id: AccountId,
+        amount: U128,
+    ) -> U128;
+}
+
+#[ext_contract(ext_fungible_token_receiver)]
+pub trait FungibleTokenReceiver {
+    fn ft_on_transfer(&mut self, sender_id: AccountId, amount: U128, msg: String) -> Promise;
```

Comment:
I think we decided on `PromiseOrValue` yes?

Reply:
For this Trait it doesn't matter

Reply:
Is it worth documenting it according to the spec, for people who implement their FT-friendly contracts by reading the source?

Reply:
It should be PromiseOrValue. 
Otherwise how does one return that it consumed X amount of tokens out of amount sent?

And yes, please add detailed doc string explaining what is happening there. Or at least linking to the standard.